CHANGELOG.md

```diff
@@ -4,6 +4,10 @@
 
 ## Important Notes
 
+- [#849](https://github.com/oauth2-proxy/oauth2-proxy/pull/849) `/oauth2/auth` `allowed_groups` querystring parameter can be paired with the `allowed-groups` configuration option.
+  - In this scenario, the user's group most be in both lists to not get a 401 response code.
+  - The `allowed_group` querystring parameter can be specified multiple times to support multiple groups.
+  - The `allowed_groups` querystring parameter can specify multiple comma delimited groups.
```

Comment:
One of these is pluralised, the other isn't, should they both be pluralised?

Reply:
The intent was to mirror our plural behavior of configurations:

If you use the singular form, you can specify it multiple times with single values in each key=value pair.

If you use the plural form, specify multiple separated by commas.

Reply:
The plural in the configuration is going away in the long term, not sure how I feel about having this here or whether we should just have `allowed_groups` and be done with it

Reply:
I'd still lean towards having this with just one option for the query string, otherwise I think this PR is fine at this point

Reply:
So go with the single comma delimited `allowed_groups`?

Reply:
I would just use the same key, so `allowed_groups=foo&allowed_groups=baz,bar` results in `[foo, bar, baz]`

I think you can do something like this for the extraction process
```
q := url.Query()
for _, allowedGroups := range q["allowed_groups"] {
  for _, group := range strings.Split(allowedGroups, ",") {
    if group != "" {
      groups[group] = struct{}{}
    }
  }
}
```

Reply:
Awesome, thanks! I missed the multi-case for `allowed_groups` when I originally ripped out `allowed_group`. Latest changes are pushed up.